parser-typechecker/src/Unison/Syntax/TypeParser.hs

```diff
@@ -55,12 +55,15 @@ type2a = delayed <|> type2
 delayed :: (Monad m, Var v) => TypeP v m
 delayed = do
   q <- reserved "'"
-  t <- effect <|> type2a
+  t <- effect <|> (pt <$> type2a)
   pure $
     Type.arrow
       (Ann (L.start q) (end $ ann t))
-      (DD.unitType (ann q))
+      (DD.thunkArgType (ann q))
       t
+  where
+    -- if no abilities listed on 't, assume '{} t
+    pt t = Type.effect (ann t) [] t
```

Comment:
This is just limiting the places where we do ability inference, since that causes round trip issues with type hashes changing. I think it's weird to use `'Nat` when you really are intending to allow effects there.